java/spring/security/injection/tainted-sql-string.yaml

```diff
@@ -52,6 +58,19 @@ rules:
                           ...
                       - pattern: $VAR += ...
                   - pattern: String.format("$SQLSTR", ...)
+                  - patterns:
+                    - pattern-inside: |
+                        String $VAR = "$SQLSTR";
+                        ...
+                    - pattern: String.format($VAR, ...)
               - metavariable-regex:
                   metavariable: $SQLSTR
                   regex: (?i)(select|delete|insert|create|update|alter|drop)\b
+    pattern-sanitizers:
+    - pattern-either:
+      - pattern: System.out.println(...)
+      - pattern: $LOG.info(...)
+      - pattern: $LOG.warning(...)
+      - pattern: $LOG.debug(...)
```

Comment:
This will incorrectly sanitize the (not uncommon) case of:

```java
string query = "SELECT * FROM users WHERE id = '" + user_id + "'";
Logger.debug("running SQL query: " + query);
...
connector.execute(query);
```

SQL injection often comes with other code smell like inappropriate logging.

Reply:
Ah, good thinking. I think I can switch the sanitizers to just use pattern-nots without adversely affecting things